docs/configuration/configuration.md

```diff
@@ -1,87 +1,166 @@
-| uri         | required `string`                                          |                                                                                                                                   |
-| provider    | required `string` (`file`, `kubernetes`, `http` or `grpc`) |                                                                                                                                   |
-| bearerToken | optional `string`                                          | Used for http sync                                                                                                                |
-| providerID  | optional `string`                                          | Value binds to grpc connection's providerID field. GRPC server implementations may use this to identify connecting flagd instance |
-| selector    | optional `string`                                          | Value binds to grpc connection's selector field. GRPC server implementations may use this to filter flag configurations           |
-| certPath    | optional `string`                                          | Used for grpcs sync when TLS certificate is needed. If not provided, system certificates will be used for TLS connection          |
+The flagd accepts a string argument, which should be a JSON representation of an array of `SourceConfig` objects.
+
+Alternatively, these configurations can be passed to flagd via config file, specified using the `--config` flag.
+
+| Field       | Type               | Note                                                                                                                                         |
+|-------------|--------------------|----------------------------------------------------------------------------------------------------------------------------------------------|
+| uri         | required `string`  | Flag configuration source of the provider                                                                                                    |
+| provider    | required `string`  | Provider type - `file`, `kubernetes`, `http` or `grpc`                                                                                       |
+| bearerToken | optional `string`  | Used for http sync and token get appended to `Authorization` header with [bearer schema](https://www.rfc-editor.org/rfc/rfc6750#section-2.1) |
+| providerID  | optional `string`  | Value binds to grpc connection's providerID field. GRPC server implementations may use this to identify connecting flagd instance            |
+| selector    | optional `string`  | Value binds to grpc connection's selector field. GRPC server implementations may use this to filter flag configurations                      |
+| grpcSecure  | optional `boolean` | Used to enable secure TLS connectivity for grpc sync. Default(ex:- if unset) is false, which will use insecure grpc connection               |
```

Comment:
Is there a reason you chose `grpcSecure` instead of something like `tls` which is more generic and might even apply to yet-to-be-added syncs?

Reply:
I don't think it needs to be done in this PR, but the HTTP sync might be improved if it was made more consistent with GRPC. It should also be able to use `tls` and `certpath` (I think). IMHO that's a reason to use `tls` right now. `certPath` is sot of the same - it's grpc specific right now but the same concept applies to HTTP.

Reply:
Agred

Reply:
Updated to use `tls` [1]

[1] - https://github.com/open-feature/flagd/blob/2a4783ef61eb3756e7a3f99aae596392906cd765/docs/configuration/configuration.md#source-configuration